### ISSUE TITLE
linter: no-restricted-imports with patterns option false negative

### ISSUE THREAD

**Issue by denis-sokolov** [oxc-project/oxc#25954]
### What version of Oxlint are you using?

1.78.0

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "rules": {
    "eslint/no-restricted-imports": ["error", { "patterns": ["foo"] } ]
  }
}
```

### What happened?

no-restricted-imports configured with a pattern `foo` only matches `foo` and not `foo/bar`.

```ts
import { foo1 } from "foo";
import { foo2 } from "foo/bar";
```

Expected: 2 errors above ([ESLint playground](https://eslint.org/play/#eyJ0ZXh0IjoiaW1wb3J0IHsgZm9vMSB9IGZyb20gXCJmb29cIjtcblxuaW1wb3J0IHsgZm9vMiB9IGZyb20gXCJmb28vMlwiO1xuXG4iLCJvcHRpb25zIjp7InJ1bGVzIjp7Im5vLXJlc3RyaWN0ZWQtaW1wb3J0cyI6WyJlcnJvciIseyJwYXR0ZXJucyI6WyJmb28iXX1dfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX0sImVjbWFWZXJzaW9uIjoyMDIxfX19)).
Actual: 1 error above.
